sno/upgrade/upgrade_v0.py

```diff
@@ -0,0 +1,265 @@
+import functools
+import json
+import re
+import subprocess
+from collections import deque
+from datetime import datetime
+from pathlib import Path
+
+import click
+import pygit2
+
+from sno import checkout, context, gpkg_adapter
+from sno.exceptions import InvalidOperation
+from sno.fast_import import fast_import_tables
+from sno.geometry import normalise_gpkg_geom
+from sno.repository_version import get_repo_version, write_repo_version_config
+from sno.schema import Schema
+from sno.structure import DatasetStructure
+
+
+def upgrade_v0(ctx, source, dest):
+    """
+    Upgrade a v0.0/v0.1 Sno repository to Sno v0.2
+    """
+    source = Path(source)
+    dest = Path(dest)
+
+    if dest.exists():
+        raise click.BadParameter(f"'{dest}': already exists", param_hint="DEST")
+
+    source_repo = pygit2.Repository(str(source))
+    if not source_repo or not source_repo.is_bare:
+        raise click.BadParameter(
+            f"'{source}': not an existing repository", param_hint="SOURCE"
+        )
```

Comment:
This boilerplate can all go in the main `upgrade` function; these versioned scripts can just take pre-validated Path objects

Reply:
Good call. There's a lot of duplicate code, rather get rid of it now than later